test/test_common/simulated_time_system_test.cc

```diff
@@ -62,6 +63,105 @@ TEST_F(SimulatedTimeSystemTest, AdvanceTimeAsync) {
   EXPECT_EQ(start_system_time_ + std::chrono::milliseconds(5), time_system_.systemTime());
 }
 
+TEST_F(SimulatedTimeSystemTest, TimerOrdering) {
+  EXPECT_EQ(start_monotonic_time_, time_system_.monotonicTime());
+  EXPECT_EQ(start_system_time_, time_system_.systemTime());
+
+  ReadyWatcher watcher1;
+  Event::TimerPtr timer1 = scheduler_->createTimer([&] { watcher1.ready(); }, dispatcher_);
+
+  ReadyWatcher watcher2;
+  Event::TimerPtr timer2 = scheduler_->createTimer([&] { watcher2.ready(); }, dispatcher_);
+
+  ReadyWatcher watcher3;
+  Event::TimerPtr timer3 = scheduler_->createTimer([&] { watcher3.ready(); }, dispatcher_);
+
+  timer1->enableTimer(std::chrono::milliseconds(0));
+  timer2->enableTimer(std::chrono::milliseconds(1));
+  timer3->enableTimer(std::chrono::milliseconds(2));
+
+  // Expect watcher calls to happen in order since timers have different times.
+  testing::InSequence s;
+  EXPECT_CALL(watcher1, ready());
+  EXPECT_CALL(watcher2, ready());
+  EXPECT_CALL(watcher3, ready());
+  advanceMsAndLoop(5);
+}
+
+// Alarms that are scheduled to execute and are cancelled do not trigger.
+TEST_F(SimulatedTimeSystemTest, TimerOrderAndDisableAlarm) {
+  EXPECT_EQ(start_monotonic_time_, time_system_.monotonicTime());
+  EXPECT_EQ(start_system_time_, time_system_.systemTime());
+
+  ReadyWatcher watcher3;
+  Event::TimerPtr timer3 = scheduler_->createTimer([&] { watcher3.ready(); }, dispatcher_);
+
+  ReadyWatcher watcher2;
+  Event::TimerPtr timer2 = scheduler_->createTimer([&] { watcher2.ready(); }, dispatcher_);
+
+  ReadyWatcher watcher1;
+  Event::TimerPtr timer1 = scheduler_->createTimer(
+      [&] {
+        timer2->disableTimer();
+        watcher1.ready();
+      },
+      dispatcher_);
+
+  timer1->enableTimer(std::chrono::milliseconds(0));
+  timer2->enableTimer(std::chrono::milliseconds(1));
+  timer3->enableTimer(std::chrono::milliseconds(2));
+
+  // Expect watcher calls to happen in order since timers have different times. watcher2 ready() is
+  // not called because timer2 was disabled as part of timer1's execution.
+  testing::InSequence s;
+  EXPECT_CALL(watcher1, ready());
+  EXPECT_CALL(watcher3, ready());
```

Comment:
I should add that this test fails without my changes.  Timer2 executes even though it was disabled.